src/Pos/Types/Core.hs

```diff
@@ -134,7 +133,9 @@ mkCoinPortion x
 -- Internally 'CoinPortion' stores 'Word64' which is divided by 'coinPortionDenominator'
 -- to get actual value. So some rounding may take place.
 unsafeCoinPortionFromDouble :: Double -> CoinPortion
-unsafeCoinPortionFromDouble x = assert (0 <= x && x <= 1) $ CoinPortion v
```

Comment:
In old projects we were using `assert`, now we usually use `panic` in such cases. I don't object.